From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jake Bailey <5341706+jakebailey@users.noreply.github.com>
Date: Fri, 14 Oct 2022 13:45:33 -0700
Subject: [PATCH] Replace all files arrays with include wildcards

Now that we are modules, the ordering of the files no longer matters.
Switch to the more typical include array with a wildcard including all
files in each project.
---
 src/cancellationToken/tsconfig.json    |   4 +-
 src/compiler/tsconfig.json             |  78 +--------
 src/debug/tsconfig.json                |   4 +-
 src/deprecatedCompat/tsconfig.json     |  13 +-
 src/executeCommandLine/tsconfig.json   |   5 +-
 src/harness/tsconfig.json              |  40 +----
 src/jsTyping/tsconfig.json             |   9 +-
 src/loggedIO/tsconfig.json             |   8 +-
 src/server/tsconfig.json               |  18 +-
 src/services/tsconfig.json             | 162 +----------------
 src/testRunner/tsconfig.json           | 230 +------------------------
 src/tsc/tsconfig.json                  |   7 +-
 src/tsserver/tsconfig.json             |  10 +-
 src/tsserverlibrary/tsconfig.json      |   8 +-
 src/typescript/tsconfig.json           |   7 +-
 src/typingsInstaller/tsconfig.json     |   7 +-
 src/typingsInstallerCore/tsconfig.json |   7 +-
 src/watchGuard/tsconfig.json           |   4 +-
 src/webServer/tsconfig.json            |   6 +-
 19 files changed, 23 insertions(+), 604 deletions(-)

diff --git a/src/cancellationToken/tsconfig.json b/src/cancellationToken/tsconfig.json
index 86550f8adf..9249fae5d1 100644
--- a/src/cancellationToken/tsconfig.json
+++ b/src/cancellationToken/tsconfig.json
@@ -9,7 +9,5 @@
             "es6"
         ]
     },
-    "files": [
-        "cancellationToken.ts"
-    ]
+    "include": ["**/*"]
 }
diff --git a/src/compiler/tsconfig.json b/src/compiler/tsconfig.json
index d8029c69e4..6b8d046414 100644
--- a/src/compiler/tsconfig.json
+++ b/src/compiler/tsconfig.json
@@ -7,81 +7,5 @@
     "references": [
     ],
 
-    "files": [
-        "corePublic.ts",
-        "core.ts",
-        "debug.ts",
-        "semver.ts",
-        "performanceCore.ts",
-        "performance.ts",
-        "perfLogger.ts",
-        "tracing.ts",
-
-        "types.ts",
-        "sys.ts",
-        "path.ts",
-        "diagnosticInformationMap.generated.ts",
-        "scanner.ts",
-        "utilitiesPublic.ts",
-        "utilities.ts",
-        "factory/baseNodeFactory.ts",
-        "factory/parenthesizerRules.ts",
-        "factory/nodeConverters.ts",
-        "factory/nodeFactory.ts",
-        "factory/emitNode.ts",
-        "factory/emitHelpers.ts",
-        "factory/nodeTests.ts",
-        "factory/utilities.ts",
-        "factory/utilitiesPublic.ts",
-        "parser.ts",
-        "commandLineParser.ts",
-        "moduleNameResolver.ts",
-
-        "binder.ts",
-        "symbolWalker.ts",
-        "checker.ts",
-        "visitorPublic.ts",
-        "sourcemap.ts",
-        "transformers/utilities.ts",
-        "transformers/destructuring.ts",
-        "transformers/taggedTemplate.ts",
-        "transformers/ts.ts",
-        "transformers/classFields.ts",
-        "transformers/typeSerializer.ts",
-        "transformers/legacyDecorators.ts",
-        "transformers/es2017.ts",
-        "transformers/es2018.ts",
-        "transformers/es2019.ts",
-        "transformers/es2020.ts",
-        "transformers/es2021.ts",
-        "transformers/esnext.ts",
-        "transformers/jsx.ts",
-        "transformers/es2016.ts",
-        "transformers/es2015.ts",
-        "transformers/es5.ts",
-        "transformers/generators.ts",
-        "transformers/module/module.ts",
-        "transformers/module/system.ts",
-        "transformers/module/esnextAnd2015.ts",
-        "transformers/module/node.ts",
-        "transformers/declarations/diagnostics.ts",
-        "transformers/declarations.ts",
-        "transformer.ts",
-        "emitter.ts",
-        "watchUtilities.ts",
-        "program.ts",
-        "builderStatePublic.ts",
-        "builderState.ts",
-        "builder.ts",
-        "builderPublic.ts",
-        "resolutionCache.ts",
-        "moduleSpecifiers.ts",
-        "watch.ts",
-        "watchPublic.ts",
-        "tsbuild.ts",
-        "tsbuildPublic.ts",
-        "_namespaces/ts.ts",
-        "_namespaces/ts.moduleSpecifiers.ts",
-        "_namespaces/ts.performance.ts"
-    ]
+    "include": ["**/*"]
 }
diff --git a/src/debug/tsconfig.json b/src/debug/tsconfig.json
index be1a75e768..ab17c2cab1 100644
--- a/src/debug/tsconfig.json
+++ b/src/debug/tsconfig.json
@@ -4,7 +4,5 @@
         "target": "es2019",
         "lib": ["es2019"],
     },
-    "files": [
-        "compilerDebug.ts",
-    ]
+    "include": ["**/*"]
 }
diff --git a/src/deprecatedCompat/tsconfig.json b/src/deprecatedCompat/tsconfig.json
index d66abe57d9..2e57445669 100644
--- a/src/deprecatedCompat/tsconfig.json
+++ b/src/deprecatedCompat/tsconfig.json
@@ -6,16 +6,5 @@
     "references": [
         { "path": "../compiler" }
     ],
-    "files": [
-        "deprecations.ts",
-        "4.0/nodeFactoryTopLevelExports.ts",
-        "4.0/renamedNodeTests.ts",
-        "4.0/renamedMapInterfaces.ts",
-        "4.2/renamedNodeTests.ts",
-        "4.2/abstractConstructorTypes.ts",
-        "4.6/importTypeAssertions.ts",
-        "4.7/typeParameterModifiers.ts",
-        "4.8/mergeDecoratorsAndModifiers.ts",
-        "_namespaces/ts.ts"
-    ]
+    "include": ["**/*"]
 }
diff --git a/src/executeCommandLine/tsconfig.json b/src/executeCommandLine/tsconfig.json
index 39a70ef314..c988aed5d1 100644
--- a/src/executeCommandLine/tsconfig.json
+++ b/src/executeCommandLine/tsconfig.json
@@ -7,8 +7,5 @@
         { "path": "../compiler" }
     ],
 
-    "files": [
-        "executeCommandLine.ts",
-        "_namespaces/ts.ts"
-    ]
+    "include": ["**/*"]
 }
diff --git a/src/harness/tsconfig.json b/src/harness/tsconfig.json
index d66c2d3be0..8349a27583 100644
--- a/src/harness/tsconfig.json
+++ b/src/harness/tsconfig.json
@@ -18,43 +18,5 @@
         { "path": "../deprecatedCompat" },
     ],
 
-    "files": [
-        "collectionsImpl.ts",
-        "util.ts",
-        "documentsUtil.ts",
-        "vpathUtil.ts",
-        "vfsUtil.ts",
-        "compilerImpl.ts",
-        "evaluatorImpl.ts",
-        "fakesHosts.ts",
-        "client.ts",
-        "findUpDir.ts",
-
-        "runnerbase.ts",
-        "sourceMapRecorder.ts",
-        "harnessGlobals.ts",
-        "harnessUtils.ts",
-        "harnessIO.ts",
-        "harnessLanguageService.ts",
-        "virtualFileSystemWithWatch.ts",
-        "fourslashImpl.ts",
-        "fourslashInterfaceImpl.ts",
-        "typeWriter.ts",
-        "_namespaces/ts.server.ts",
-        "_namespaces/ts.ts",
-        "_namespaces/collections.ts",
-        "_namespaces/compiler.ts",
-        "_namespaces/documents.ts",
-        "_namespaces/evaluator.ts",
-        "_namespaces/fakes.ts",
-        "_namespaces/Utils.ts",
-        "_namespaces/FourSlash.ts",
-        "_namespaces/FourSlashInterface.ts",
-        "_namespaces/Harness.ts",
-        "_namespaces/Harness.LanguageService.ts",
-        "_namespaces/Harness.SourceMapRecorder.ts",
-        "_namespaces/vfs.ts",
-        "_namespaces/ts.TestFSWithWatch.ts",
-        "_namespaces/vpath.ts"
-    ]
+    "include": ["**/*"]
 }
diff --git a/src/jsTyping/tsconfig.json b/src/jsTyping/tsconfig.json
index d28ea253a8..048e343779 100644
--- a/src/jsTyping/tsconfig.json
+++ b/src/jsTyping/tsconfig.json
@@ -12,12 +12,5 @@
     "references": [
         { "path": "../compiler" }
     ],
-    "files": [
-        "shared.ts",
-        "types.ts",
-        "jsTyping.ts",
-        "_namespaces/ts.JsTyping.ts",
-        "_namespaces/ts.ts",
-        "_namespaces/ts.server.ts"
-    ]
+    "include": ["**/*"]
 }
diff --git a/src/loggedIO/tsconfig.json b/src/loggedIO/tsconfig.json
index 9829567db5..820cd1cb57 100644
--- a/src/loggedIO/tsconfig.json
+++ b/src/loggedIO/tsconfig.json
@@ -17,11 +17,5 @@
         { "path": "../harness" },
     ],
 
-    "files": [
-        "loggedIO.ts",
-        "_namespaces/Playback.ts",
-        "_namespaces/ts.server.ts",
-        "_namespaces/ts.ts",
-        "_namespaces/Harness.ts"
-    ]
+    "include": ["**/*"]
 }
diff --git a/src/server/tsconfig.json b/src/server/tsconfig.json
index 674ed67018..3ab8107c3e 100644
--- a/src/server/tsconfig.json
+++ b/src/server/tsconfig.json
@@ -10,21 +10,5 @@
         { "path": "../jsTyping" },
         { "path": "../services" }
     ],
-    "files": [
-        "types.ts",
-        "utilitiesPublic.ts",
-        "utilities.ts",
-        "protocol.ts",
-        "scriptInfo.ts",
-        "typingsCache.ts",
-        "project.ts",
-        "editorServices.ts",
-        "moduleSpecifierCache.ts",
-        "packageJsonCache.ts",
-        "session.ts",
-        "scriptVersionCache.ts",
-        "_namespaces/ts.server.ts",
-        "_namespaces/ts.ts",
-        "_namespaces/ts.server.protocol.ts"
-    ]
+    "include": ["**/*"]
 }
diff --git a/src/services/tsconfig.json b/src/services/tsconfig.json
index 4a3e47d443..42c124fdd8 100644
--- a/src/services/tsconfig.json
+++ b/src/services/tsconfig.json
@@ -6,165 +6,5 @@
         { "path": "../compiler" },
         { "path": "../jsTyping" }
     ],
-    "files": [
-        "types.ts",
-        "utilities.ts",
-        "exportInfoMap.ts",
-        "classifier.ts",
-        "classifier2020.ts",
-        "stringCompletions.ts",
-        "completions.ts",
-        "documentHighlights.ts",
-        "documentRegistry.ts",
-        "importTracker.ts",
-        "findAllReferences.ts",
-        "callHierarchy.ts",
-        "getEditsForFileRename.ts",
-        "goToDefinition.ts",
-        "jsDoc.ts",
-        "navigateTo.ts",
-        "navigationBar.ts",
-        "organizeImports.ts",
-        "getEditsForFileRename.ts",
-        "outliningElementsCollector.ts",
-        "patternMatcher.ts",
-        "preProcess.ts",
-        "rename.ts",
-        "smartSelection.ts",
-        "signatureHelp.ts",
-        "inlayHints.ts",
-        "sourcemaps.ts",
-        "suggestionDiagnostics.ts",
-        "symbolDisplay.ts",
-        "transpile.ts",
-        "formatting/formattingContext.ts",
-        "formatting/formattingScanner.ts",
-        "formatting/rule.ts",
-        "formatting/rules.ts",
-        "formatting/rulesMap.ts",
-        "formatting/formatting.ts",
-        "formatting/smartIndenter.ts",
-        "textChanges.ts",
-        "codeFixProvider.ts",
-        "refactorProvider.ts",
-        "codefixes/addConvertToUnknownForNonOverlappingTypes.ts",
-        "codefixes/addEmptyExportDeclaration.ts",
-        "codefixes/addMissingAsync.ts",
-        "codefixes/addMissingAwait.ts",
-        "codefixes/addMissingConst.ts",
-        "codefixes/addMissingDeclareProperty.ts",
-        "codefixes/addMissingInvocationForDecorator.ts",
-        "codefixes/addNameToNamelessParameter.ts",
-        "codefixes/addOptionalPropertyUndefined.ts",
-        "codefixes/annotateWithTypeFromJSDoc.ts",
-        "codefixes/convertFunctionToEs6Class.ts",
-        "codefixes/convertToAsyncFunction.ts",
-        "codefixes/convertToEsModule.ts",
-        "codefixes/correctQualifiedNameToIndexedAccessType.ts",
-        "codefixes/convertToTypeOnlyExport.ts",
-        "codefixes/convertToTypeOnlyImport.ts",
-        "codefixes/convertLiteralTypeToMappedType.ts",
-        "codefixes/fixClassIncorrectlyImplementsInterface.ts",
-        "codefixes/importFixes.ts",
-        "codefixes/fixAddMissingConstraint.ts",
-        "codefixes/fixOverrideModifier.ts",
-        "codefixes/fixNoPropertyAccessFromIndexSignature.ts",
-        "codefixes/fixImplicitThis.ts",
-        "codefixes/fixImportNonExportedMember.ts",
-        "codefixes/fixIncorrectNamedTupleSyntax.ts",
-        "codefixes/fixSpelling.ts",
-        "codefixes/returnValueCorrect.ts",
-        "codefixes/fixAddMissingMember.ts",
-        "codefixes/fixAddMissingNewOperator.ts",
-        "codefixes/fixCannotFindModule.ts",
-        "codefixes/fixClassDoesntImplementInheritedAbstractMember.ts",
-        "codefixes/fixClassSuperMustPrecedeThisAccess.ts",
-        "codefixes/fixConstructorForDerivedNeedSuperCall.ts",
-        "codefixes/fixEnableExperimentalDecorators.ts",
-        "codefixes/fixEnableJsxFlag.ts",
-        "codefixes/fixNaNEquality.ts",
-        "codefixes/fixModuleAndTargetOptions.ts",
-        "codefixes/fixPropertyAssignment.ts",
-        "codefixes/fixExtendsInterfaceBecomesImplements.ts",
-        "codefixes/fixForgottenThisPropertyAccess.ts",
-        "codefixes/fixInvalidJsxCharacters.ts",
-        "codefixes/fixUnmatchedParameter.ts",
-        "codefixes/fixUnreferenceableDecoratorMetadata.ts",
-        "codefixes/fixUnusedIdentifier.ts",
-        "codefixes/fixUnreachableCode.ts",
-        "codefixes/fixUnusedLabel.ts",
-        "codefixes/fixJSDocTypes.ts",
-        "codefixes/fixMissingCallParentheses.ts",
-        "codefixes/fixAwaitInSyncFunction.ts",
-        "codefixes/fixPropertyOverrideAccessor.ts",
-        "codefixes/inferFromUsage.ts",
-        "codefixes/fixReturnTypeInAsyncFunction.ts",
-        "codefixes/disableJsDiagnostics.ts",
-        "codefixes/helpers.ts",
-        "codefixes/generateAccessors.ts",
-        "codefixes/fixInvalidImportSyntax.ts",
-        "codefixes/fixStrictClassInitialization.ts",
-        "codefixes/requireInTs.ts",
-        "codefixes/useDefaultImport.ts",
-        "codefixes/useBigintLiteral.ts",
-        "codefixes/fixAddModuleReferTypeMissingTypeof.ts",
-        "codefixes/wrapJsxInFragment.ts",
-        "codefixes/convertToMappedObjectType.ts",
-        "codefixes/removeAccidentalCallParentheses.ts",
-        "codefixes/removeUnnecessaryAwait.ts",
-        "codefixes/splitTypeOnlyImport.ts",
-        "codefixes/convertConstToLet.ts",
-        "codefixes/fixExpectedComma.ts",
-        "codefixes/fixAddVoidToPromise.ts",
-        "refactors/convertExport.ts",
-        "refactors/convertImport.ts",
-        "refactors/convertToOptionalChainExpression.ts",
-        "refactors/convertOverloadListToSingleSignature.ts",
-        "refactors/extractSymbol.ts",
-        "refactors/extractType.ts",
-        "refactors/generateGetAccessorAndSetAccessor.ts",
-        "refactors/helpers.ts",
-        "refactors/moveToNewFile.ts",
-        "refactors/addOrRemoveBracesToArrowFunction.ts",
-        "refactors/convertParamsToDestructuredObject.ts",
-        "refactors/convertStringOrTemplateLiteral.ts",
-        "refactors/convertArrowFunctionOrFunctionExpression.ts",
-        "refactors/inferFunctionReturnType.ts",
-        "services.ts",
-        "breakpoints.ts",
-        "transform.ts",
-        "shims.ts",
-        "globalThisShim.ts",
-        "_namespaces/ts.BreakpointResolver.ts",
-        "_namespaces/ts.ts",
-        "_namespaces/ts.CallHierarchy.ts",
-        "_namespaces/ts.classifier.v2020.ts",
-        "_namespaces/ts.classifier.ts",
-        "_namespaces/ts.codefix.ts",
-        "_namespaces/ts.Completions.ts",
-        "_namespaces/ts.FindAllReferences.ts",
-        "_namespaces/ts.GoToDefinition.ts",
-        "_namespaces/ts.InlayHints.ts",
-        "_namespaces/ts.JsDoc.ts",
-        "_namespaces/ts.NavigateTo.ts",
-        "_namespaces/ts.NavigationBar.ts",
-        "_namespaces/ts.OrganizeImports.ts",
-        "_namespaces/ts.OutliningElementsCollector.ts",
-        "_namespaces/ts.refactor.ts",
-        "_namespaces/ts.Rename.ts",
-        "_namespaces/ts.SignatureHelp.ts",
-        "_namespaces/ts.SmartSelectionRange.ts",
-        "_namespaces/ts.Completions.StringCompletions.ts",
-        "_namespaces/ts.SymbolDisplay.ts",
-        "_namespaces/ts.textChanges.ts",
-        "_namespaces/ts.formatting.ts",
-        "_namespaces/ts.refactor.addOrRemoveBracesToArrowFunction.ts",
-        "_namespaces/ts.refactor.convertArrowFunctionOrFunctionExpression.ts",
-        "_namespaces/ts.refactor.convertParamsToDestructuredObject.ts",
-        "_namespaces/ts.refactor.convertStringOrTemplateLiteral.ts",
-        "_namespaces/ts.refactor.convertToOptionalChainExpression.ts",
-        "_namespaces/ts.refactor.extractSymbol.ts",
-        "_namespaces/ts.refactor.generateGetAccessorAndSetAccessor.ts",
-        "_namespaces/ts.refactor.inferFunctionReturnType.ts"
-    ]
+    "include": ["**/*"]
 }
diff --git a/src/testRunner/tsconfig.json b/src/testRunner/tsconfig.json
index de98074792..2b8b1842ce 100644
--- a/src/testRunner/tsconfig.json
+++ b/src/testRunner/tsconfig.json
@@ -22,233 +22,5 @@
         { "path": "../loggedIO" }
     ],
 
-    "files": [
-        "fourslashRunner.ts",
-        "compilerRunner.ts",
-        "projectsRunner.ts",
-        "rwcRunner.ts",
-        "externalCompileRunner.ts",
-        "test262Runner.ts",
-
-        "parallel/host.ts",
-        "parallel/worker.ts",
-        "parallel/shared.ts",
-
-        "runner.ts",
-
-        "unittests/helpers.ts",
-        "unittests/services/extract/helpers.ts",
-        "unittests/tsbuild/helpers.ts",
-        "unittests/tsc/helpers.ts",
-        "unittests/tscWatch/helpers.ts",
-        "unittests/tsserver/helpers.ts",
-
-        "unittests/asserts.ts",
-        "unittests/base64.ts",
-        "unittests/builder.ts",
-        "unittests/comments.ts",
-        "unittests/compilerCore.ts",
-        "unittests/convertToBase64.ts",
-        "unittests/customTransforms.ts",
-        "unittests/factory.ts",
-        "unittests/incrementalParser.ts",
-        "unittests/jsDocParsing.ts",
-        "unittests/jsonParserRecovery.ts",
-        "unittests/moduleResolution.ts",
-        "unittests/parsePseudoBigInt.ts",
-        "unittests/paths.ts",
-        "unittests/printer.ts",
-        "unittests/programApi.ts",
-        "unittests/publicApi.ts",
-        "unittests/reuseProgramStructure.ts",
-        "unittests/semver.ts",
-        "unittests/transform.ts",
-        "unittests/typeParameterIsPossiblyReferenced.ts",
-        "unittests/config/commandLineParsing.ts",
-        "unittests/config/configurationExtension.ts",
-        "unittests/config/convertCompilerOptionsFromJson.ts",
-        "unittests/config/convertTypeAcquisitionFromJson.ts",
-        "unittests/config/initializeTSConfig.ts",
-        "unittests/config/matchFiles.ts",
-        "unittests/config/projectReferences.ts",
-        "unittests/config/showConfig.ts",
-        "unittests/config/tsconfigParsing.ts",
-        "unittests/config/tsconfigParsingWatchOptions.ts",
-        "unittests/evaluation/arraySpread.ts",
-        "unittests/evaluation/asyncArrow.ts",
-        "unittests/evaluation/asyncGenerator.ts",
-        "unittests/evaluation/autoAccessors.ts",
-        "unittests/evaluation/awaiter.ts",
-        "unittests/evaluation/destructuring.ts",
-        "unittests/evaluation/externalModules.ts",
-        "unittests/evaluation/forAwaitOf.ts",
-        "unittests/evaluation/forOf.ts",
-        "unittests/evaluation/optionalCall.ts",
-        "unittests/evaluation/objectRest.ts",
-        "unittests/evaluation/superInStaticInitializer.ts",
-        "unittests/evaluation/templateLiteral.ts",
-        "unittests/evaluation/updateExpressionInModule.ts",
-        "unittests/services/cancellableLanguageServiceOperations.ts",
-        "unittests/services/colorization.ts",
-        "unittests/services/convertToAsyncFunction.ts",
-        "unittests/services/documentRegistry.ts",
-        "unittests/services/extract/constants.ts",
-        "unittests/services/extract/functions.ts",
-        "unittests/services/extract/symbolWalker.ts",
-        "unittests/services/extract/ranges.ts",
-        "unittests/services/hostNewLineSupport.ts",
-        "unittests/services/languageService.ts",
-        "unittests/services/organizeImports.ts",
-        "unittests/services/patternMatcher.ts",
-        "unittests/services/preProcessFile.ts",
-        "unittests/services/textChanges.ts",
-        "unittests/services/transpile.ts",
-        "unittests/tsbuild/amdModulesWithOut.ts",
-        "unittests/tsbuild/clean.ts",
-        "unittests/tsbuild/configFileErrors.ts",
-        "unittests/tsbuild/configFileExtends.ts",
-        "unittests/tsbuild/containerOnlyReferenced.ts",
-        "unittests/tsbuild/declarationEmit.ts",
-        "unittests/tsbuild/demo.ts",
-        "unittests/tsbuild/emitDeclarationOnly.ts",
-        "unittests/tsbuild/emptyFiles.ts",
-        "unittests/tsbuild/exitCodeOnBogusFile.ts",
-        "unittests/tsbuild/graphOrdering.ts",
-        "unittests/tsbuild/inferredTypeFromTransitiveModule.ts",
-        "unittests/tsbuild/javascriptProjectEmit.ts",
-        "unittests/tsbuild/lateBoundSymbol.ts",
-        "unittests/tsbuild/moduleResolution.ts",
-        "unittests/tsbuild/moduleSpecifiers.ts",
-        "unittests/tsbuild/noEmit.ts",
-        "unittests/tsbuild/noEmitOnError.ts",
-        "unittests/tsbuild/outFile.ts",
-        "unittests/tsbuild/outputPaths.ts",
-        "unittests/tsbuild/publicApi.ts",
-        "unittests/tsbuild/referencesWithRootDirInParent.ts",
-        "unittests/tsbuild/resolveJsonModule.ts",
-        "unittests/tsbuild/sample.ts",
-        "unittests/tsbuild/transitiveReferences.ts",
-        "unittests/tsbuildWatch/configFileErrors.ts",
-        "unittests/tsbuildWatch/demo.ts",
-        "unittests/tsbuildWatch/moduleResolution.ts",
-        "unittests/tsbuildWatch/noEmit.ts",
-        "unittests/tsbuildWatch/noEmitOnError.ts",
-        "unittests/tsbuildWatch/programUpdates.ts",
-        "unittests/tsbuildWatch/projectsBuilding.ts",
-        "unittests/tsbuildWatch/publicApi.ts",
-        "unittests/tsbuildWatch/reexport.ts",
-        "unittests/tsbuildWatch/watchEnvironment.ts",
-        "unittests/tsc/cancellationToken.ts",
-        "unittests/tsc/composite.ts",
-        "unittests/tsc/declarationEmit.ts",
-        "unittests/tsc/forceConsistentCasingInFileNames.ts",
-        "unittests/tsc/incremental.ts",
-        "unittests/tsc/listFilesOnly.ts",
-        "unittests/tsc/projectReferences.ts",
-        "unittests/tsc/redirect.ts",
-        "unittests/tsc/runWithoutArgs.ts",
-        "unittests/tscWatch/consoleClearing.ts",
-        "unittests/tscWatch/emit.ts",
-        "unittests/tscWatch/nodeNextWatch.ts",
-        "unittests/tscWatch/emitAndErrorUpdates.ts",
-        "unittests/tscWatch/forceConsistentCasingInFileNames.ts",
-        "unittests/tscWatch/incremental.ts",
-        "unittests/tscWatch/moduleResolution.ts",
-        "unittests/tscWatch/programUpdates.ts",
-        "unittests/tscWatch/projectsWithReferences.ts",
-        "unittests/tscWatch/resolutionCache.ts",
-        "unittests/tscWatch/sourceOfProjectReferenceRedirect.ts",
-        "unittests/tscWatch/watchApi.ts",
-        "unittests/tscWatch/watchEnvironment.ts",
-        "unittests/tsserver/applyChangesToOpenFiles.ts",
-        "unittests/tsserver/autoImportProvider.ts",
-        "unittests/tsserver/auxiliaryProject.ts",
-        "unittests/tsserver/cachingFileSystemInformation.ts",
-        "unittests/tsserver/cancellationToken.ts",
-        "unittests/tsserver/compileOnSave.ts",
-        "unittests/tsserver/completions.ts",
-        "unittests/tsserver/completionsIncomplete.ts",
-        "unittests/tsserver/configFileSearch.ts",
-        "unittests/tsserver/configuredProjects.ts",
-        "unittests/tsserver/declarationFileMaps.ts",
-        "unittests/tsserver/documentRegistry.ts",
-        "unittests/tsserver/duplicatePackages.ts",
-        "unittests/tsserver/dynamicFiles.ts",
-        "unittests/tsserver/events/largeFileReferenced.ts",
-        "unittests/tsserver/events/projectLanguageServiceState.ts",
-        "unittests/tsserver/events/projectLoading.ts",
-        "unittests/tsserver/events/projectUpdatedInBackground.ts",
-        "unittests/tsserver/exportMapCache.ts",
-        "unittests/tsserver/externalProjects.ts",
-        "unittests/tsserver/forceConsistentCasingInFileNames.ts",
-        "unittests/tsserver/formatSettings.ts",
-        "unittests/tsserver/getApplicableRefactors.ts",
-        "unittests/tsserver/getEditsForFileRename.ts",
-        "unittests/tsserver/getExportReferences.ts",
-        "unittests/tsserver/getFileReferences.ts",
-        "unittests/tsserver/importHelpers.ts",
-        "unittests/tsserver/inlayHints.ts",
-        "unittests/tsserver/inferredProjects.ts",
-        "unittests/tsserver/jsdocTag.ts",
-        "unittests/tsserver/languageService.ts",
-        "unittests/tsserver/maxNodeModuleJsDepth.ts",
-        "unittests/tsserver/metadataInResponse.ts",
-        "unittests/tsserver/moduleResolution.ts",
-        "unittests/tsserver/moduleSpecifierCache.ts",
-        "unittests/tsserver/navTo.ts",
-        "unittests/tsserver/occurences.ts",
-        "unittests/tsserver/openFile.ts",
-        "unittests/tsserver/packageJsonInfo.ts",
-        "unittests/tsserver/partialSemanticServer.ts",
-        "unittests/tsserver/plugins.ts",
-        "unittests/tsserver/projectErrors.ts",
-        "unittests/tsserver/projectReferenceCompileOnSave.ts",
-        "unittests/tsserver/projectReferenceErrors.ts",
-        "unittests/tsserver/projectReferences.ts",
-        "unittests/tsserver/projectReferencesSourcemap.ts",
-        "unittests/tsserver/projects.ts",
-        "unittests/tsserver/projectsWithReferences.ts",
-        "unittests/tsserver/refactors.ts",
-        "unittests/tsserver/reload.ts",
-        "unittests/tsserver/reloadProjects.ts",
-        "unittests/tsserver/rename.ts",
-        "unittests/tsserver/resolutionCache.ts",
-        "unittests/tsserver/session.ts",
-        "unittests/tsserver/skipLibCheck.ts",
-        "unittests/tsserver/smartSelection.ts",
-        "unittests/tsserver/symlinkCache.ts",
-        "unittests/tsserver/symLinks.ts",
-        "unittests/tsserver/syntacticServer.ts",
-        "unittests/tsserver/syntaxOperations.ts",
-        "unittests/tsserver/textStorage.ts",
-        "unittests/tsserver/telemetry.ts",
-        "unittests/tsserver/typeAquisition.ts",
-        "unittests/tsserver/typeOnlyImportChains.ts",
-        "unittests/tsserver/typeReferenceDirectives.ts",
-        "unittests/tsserver/typingsInstaller.ts",
-        "unittests/tsserver/versionCache.ts",
-        "unittests/tsserver/watchEnvironment.ts",
-        "unittests/tsserver/webServer.ts",
-        "unittests/debugDeprecation.ts",
-        "_namespaces/compiler.ts",
-        "_namespaces/Harness.ts",
-        "_namespaces/documents.ts",
-        "_namespaces/evaluator.ts",
-        "_namespaces/fakes.ts",
-        "_namespaces/FourSlash.ts",
-        "_namespaces/Playback.ts",
-        "_namespaces/project.ts",
-        "_namespaces/RWC.ts",
-        "_namespaces/Utils.ts",
-        "_namespaces/vfs.ts",
-        "_namespaces/vpath.ts",
-        "_namespaces/Harness.Parallel.Host.ts",
-        "_namespaces/Harness.Parallel.ts",
-        "_namespaces/Harness.Parallel.Worker.ts",
-        "_namespaces/ts.ts",
-        "_namespaces/ts.tscWatch.ts",
-        "_namespaces/ts.projectSystem.ts",
-        "_namespaces/ts.server.ts",
-        "_namespaces/tests.ts"
-    ]
+    "include": ["**/*"]
 }
diff --git a/src/tsc/tsconfig.json b/src/tsc/tsconfig.json
index b4e1a49585..21e193ad5d 100644
--- a/src/tsc/tsconfig.json
+++ b/src/tsc/tsconfig.json
@@ -2,12 +2,9 @@
     "extends": "../tsconfig-base",
     "compilerOptions": {
     },
-    "files": [
-        "tsc.ts",
-        "_namespaces/ts.ts"
-    ],
     "references": [
         { "path": "../compiler" },
         { "path": "../executeCommandLine" }
-    ]
+    ],
+    "include": ["**/*"]
 }
diff --git a/src/tsserver/tsconfig.json b/src/tsserver/tsconfig.json
index c2508ba284..8408638400 100644
--- a/src/tsserver/tsconfig.json
+++ b/src/tsserver/tsconfig.json
@@ -6,13 +6,6 @@
             "node"
         ]
     },
-    "files": [
-        "nodeServer.ts",
-        "webServer.ts",
-        "server.ts",
-        "_namespaces/ts.server.ts",
-        "_namespaces/ts.ts"
-    ],
     "references": [
         { "path": "../compiler" },
         { "path": "../services" },
@@ -20,5 +13,6 @@
         { "path": "../server" },
         { "path": "../webServer" },
         { "path": "../deprecatedCompat" }
-    ]
+    ],
+    "include": ["**/*"]
 }
diff --git a/src/tsserverlibrary/tsconfig.json b/src/tsserverlibrary/tsconfig.json
index a261c45cd2..97af0d9529 100644
--- a/src/tsserverlibrary/tsconfig.json
+++ b/src/tsserverlibrary/tsconfig.json
@@ -2,16 +2,12 @@
     "extends": "../tsconfig-base",
     "compilerOptions": {
     },
-    "files": [
-        "tsserverlibrary.ts",
-        "_namespaces/ts.server.ts",
-        "_namespaces/ts.ts"
-    ],
     "references": [
         { "path": "../compiler" },
         { "path": "../jsTyping" },
         { "path": "../services" },
         { "path": "../server" },
         { "path": "../deprecatedCompat" }
-    ]
+    ],
+    "include": ["**/*"]
 }
diff --git a/src/typescript/tsconfig.json b/src/typescript/tsconfig.json
index 816a2a37a9..4bc23cc65b 100644
--- a/src/typescript/tsconfig.json
+++ b/src/typescript/tsconfig.json
@@ -2,14 +2,11 @@
     "extends": "../tsconfig-base",
     "compilerOptions": {
     },
-    "files": [
-        "typescript.ts",
-        "_namespaces/ts.ts"
-    ],
     "references": [
         { "path": "../compiler" },
         { "path": "../jsTyping" },
         { "path": "../services" },
         { "path": "../deprecatedCompat" }
-    ]
+    ],
+    "include": ["**/*"]
 }
diff --git a/src/typingsInstaller/tsconfig.json b/src/typingsInstaller/tsconfig.json
index e197ac5bfa..528f3580c1 100644
--- a/src/typingsInstaller/tsconfig.json
+++ b/src/typingsInstaller/tsconfig.json
@@ -14,10 +14,5 @@
         { "path": "../jsTyping" },
         { "path": "../typingsInstallerCore" }
     ],
-    "files": [
-        "nodeTypingsInstaller.ts",
-        "_namespaces/ts.server.typingsInstaller.ts",
-        "_namespaces/ts.ts",
-        "_namespaces/ts.server.ts"
-    ]
+    "include": ["**/*"]
 }
diff --git a/src/typingsInstallerCore/tsconfig.json b/src/typingsInstallerCore/tsconfig.json
index 94c333f087..83fabc202f 100644
--- a/src/typingsInstallerCore/tsconfig.json
+++ b/src/typingsInstallerCore/tsconfig.json
@@ -13,10 +13,5 @@
         { "path": "../compiler" },
         { "path": "../jsTyping" }
     ],
-    "files": [
-        "typingsInstaller.ts",
-        "_namespaces/ts.server.typingsInstaller.ts",
-        "_namespaces/ts.ts",
-        "_namespaces/ts.server.ts"
-    ]
+    "include": ["**/*"]
 }
diff --git a/src/watchGuard/tsconfig.json b/src/watchGuard/tsconfig.json
index 0067737ae2..636a871a94 100644
--- a/src/watchGuard/tsconfig.json
+++ b/src/watchGuard/tsconfig.json
@@ -8,7 +8,5 @@
             "es6"
         ]
     },
-    "files": [
-        "watchGuard.ts"
-    ]
+    "include": ["**/*"]
 }
diff --git a/src/webServer/tsconfig.json b/src/webServer/tsconfig.json
index 89b95927dd..5b4c1fdbf5 100644
--- a/src/webServer/tsconfig.json
+++ b/src/webServer/tsconfig.json
@@ -11,9 +11,5 @@
         { "path": "../services" },
         { "path": "../server" }
     ],
-    "files": [
-        "webServer.ts",
-        "_namespaces/ts.server.ts",
-        "_namespaces/ts.ts"
-    ]
+    "include": ["**/*"]
 }
-- 
2.38.0

